### PR TITLE
ARROW-9693: [CI][Docs] Nightly docs build fails

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs

File: ci/conda_env_sphinx.yml [hunk 1]
```diff
@@ -19,6 +19,6 @@
 breathe
 doxygen
 ipython
-# Pinned per ARROW-8340
-sphinx=2.4.4
+# Pinned per ARROW-9693
+sphinx=3.1
```
Comment (pitrou) [inline ci/conda_env_sphinx.yml:23]: 3.1.2?

Comment (kou) [inline ci/conda_env_sphinx.yml:23]:
```suggestion
sphinx=3.1.2
```

File: ci/docker/linux-apt-docs.dockerfile [hunk 2]
```diff
@@ -68,11 +68,12 @@ RUN wget -q -O - https://deb.nodesource.com/setup_${node}.x | bash - && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
+# Sphinx is pinned because of ARROW-9693
 RUN pip install \
         meson \
         breathe \
         ipython \
-        sphinx \
+        sphinx==3.1 \
```
Comment (kou) [inline ci/docker/linux-apt-docs.dockerfile:76]:
```suggestion
        sphinx==3.1.2 \
```

**Comment by kou** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**